### PR TITLE
Convert ./components/file_info_preview/file_info_preview.tsx from Class Component to Function Component

### PR DESCRIPTION
**Pull request by dipaksinha1**
Summary

Refactors webapp\channels\src\components\file_info_preview\file_info_preview.tsx from Class Component to Function Component

Details of the changes within the comments of the PR.

Ticket Link

Issue Number: #26213 

Release Note
```release-note
NONE
```

[MM-56851]: https://mattermost.atlassian.net/browse/MM-56851?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dipaksinha1** [comment]
check-cla

File: webapp/channels/src/components/file_info_preview/file_info_preview.tsx [hunk 1]
```diff
@@ -13,58 +13,61 @@ type Props = {
     canDownloadFiles: boolean;
 };
 
-export default class FileInfoPreview extends React.PureComponent<Props> {
-    render() {
-        const fileInfo = this.props.fileInfo;
-        const fileUrl = this.props.fileUrl;
+const FileInfoPreview = ({fileInfo, fileUrl, canDownloadFiles}: Props) => {
```
Comment (larkox) [inline webapp/channels/src/components/file_info_preview/file_info_preview.tsx:16]: Nit: Can you put every prop in a different line? As in:
```suggestion
const FileInfoPreview = ({
    fileInfo,
    fileUrl,
    canDownloadFiles,
}: Props) => {
```

This way, if we ever make any change to this component, is way more visible what props are removed / added.

**Review by larkox — APPROVED** [review]
Great work! Just one style nit.

File: webapp/channels/src/components/file_info_preview/file_info_preview.tsx (hunk 1 above)
Comment (dipaksinha1) [reply webapp/channels/src/components/file_info_preview/file_info_preview.tsx]: Got it, Thanks

**Comment by hmhealey** [comment]
Thanks for the PR, @dipaksinha1. Would you mind fixing the failing tests for this? You can run those locally by running `npm run test`.

Since the failures look like they're just outdated snapshots, you should be able to update those by running `npm run test -- -u`

**Comment by dipaksinha1** [comment]
Hi @hmhealey , I have updated the snapshot. I request you to please review and let me know if any changes are required.

**Review by hmhealey — APPROVED** [review]
Yep, that looks good. I'll just update the branch to fix the smoke test issue which is caused by upstream changes

/update-branch

**Pull request merged**